rust/storage/src/local.rs

```diff
@@ -19,7 +24,47 @@ impl LocalStorage {
         };
     }
 
-    pub async fn get(
+    pub async fn get(&self, key: &str) -> Result<Arc<Vec<u8>>, String> {
+        let mut stream = self
+            .get_stream(&key)
+            .instrument(tracing::trace_span!(parent: Span::current(), "Local Storage get"))
+            .await?;
+        let read_block_span =
+            tracing::trace_span!(parent: Span::current(), "Local storage read bytes to end");
+        let buf = read_block_span
+            .in_scope(|| async {
+                let mut buf: Vec<u8> = Vec::new();
```

Comment:
note: for non-local it might make sense to preallocate this to be max block size

Reply:
The max block size needs to be propagated from the block manager. Is it ok to do it in a separate PR given that it is net new improvement which does not exist even in base?

Reply:
Added to tracker and discussed offline that it is ok to do it later